python/dcr_template.py

```diff
@@ -39,61 +40,132 @@
 import lsst.meas.algorithms as measAlg
 import lsst.pex.exceptions
 import lsst.pex.policy as pexPolicy
-from lsst.sims.photUtils import Bandpass, PhotometricParameters
+from lsst.sims.photUtils import Bandpass
 from lsst.utils import getPackageDir
 
 from .calc_refractive_index import diff_refraction
 
-__all__ = ["DcrModel", "DcrCorrection"]
+
+__all__ = ("DcrModel", "DcrCorrection")
 
 nanFloat = float("nan")
 nanAngle = Angle(nanFloat)
 lsst_lat = Angle(np.radians(-30.244639))
 lsst_lon = Angle(np.radians(-70.749417))
 lsst_alt = 2663.
 
+# Temporary debugging parameters, used if debug_mode=True or self.debug=True is set.
+x0 = 300
+dx = 200
+y0 = 500
+dy = 200
+
 
 class DcrModel:
-    """!Lightweight object with only the minimum needed to generate DCR-matched template exposures."""
+    """!Lightweight object with only the minimum needed to generate DCR-matched template exposures.
+
+    A model must first be generated with DcrCorrection (below). That model can then be used directly, or
+    persisted and later read back in needing only this more lightweight class that doesn't need to carry
+    around the input exposures. This class will generate template exposures suitable for
+    image differencing that are matched to existing exposures in a repository.
+
+    Attributes
+    ----------
+    bandpass : lsst.sims.photUtils.Bandpass object
+        Bandpass object returned by load_bandpass
+    bbox : lsst.afw.geom.Box2I object
+        A bounding box.
+    butler : lsst.daf.persistence Butler object
+        The butler handles persisting and depersisting data to and from a repository.
+    debug : bool
+        Temporary debugging option.
+        If set, calculations are performed on only a small region of the full images.
+    detected_bit : int
+        Value of the detected bit in the bit plane mask.
+    exposure_time : float
+        Length of the exposure, in seconds.
+    filter_name : str
+        Name of the bandpass-defining filter of the data. Expected values are u,g,r,i,z,y.
+    instrument : str
+        Name of the observatory. Used to format dataIds for the butler.
+    mask : np.ndarray
+        Combined bit plane mask of the model, which is used as the mask plane for generated templates.
+    model : list of np.ndarrays
+        The DCR model to be used to generate templates. Contains one array for each wavelength step.
+    n_step : int
+        Number of sub-filter wavelength planes to model.
+    pixel_scale : float
+        Plate scale, in arcseconds.
+    psf : lsst.meas.algorithms KernelPsf object
+        Representation of the point spread function (PSF) of the model.
+    psf_size : int
+        Dimension of the PSF, in pixels.
+    wcs : lsst.afw.image Wcs object
+        World Coordinate System of the model.
+    weights : np.ndarray
+        Weights of the model, calculated from the combined inverse variance of the input exposures.
```

Comment:
I've tried to clarify the explanation of this and the previous two parameters you commented on.